tests/BasicSearchTests.FunctionalTests.Core/TestSupport/V3SearchBuilder.cs

```diff
@@ -1,12 +1,42 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
+using System.Collections.Specialized;
 
 namespace BasicSearchTests.FunctionalTests.Core.TestSupport
 {
     public class V3SearchBuilder : QueryBuilder
     {
```

Comment:
A V3 search client already exists in the Azure Search functional tests, can we converge these? See: https://github.com/NuGet/NuGet.Services.Metadata/blob/master/tests/NuGet.Services.AzureSearch.FunctionalTests/Support/NuGetSearchFunctionalTestBase.cs#L43

Reply:
Idea: if/when you converge these client libraries, you could create a new `NuGet.Protocol.Search` library that follows the [`NuGet.Protocol.Catalog`](https://github.com/NuGet/NuGet.Services.Metadata/tree/master/src/NuGet.Protocol.Catalog) pattern. I would move the search service [request/response models](https://github.com/NuGet/NuGet.Services.Metadata/tree/master/src/NuGet.Services.AzureSearch/SearchService/Models) to this new project and have your client library implement [`ISearchService`](https://github.com/NuGet/NuGet.Services.Metadata/blob/master/src/NuGet.Services.AzureSearch/SearchService/ISearchService.cs).

For example: [`SearchClient.SearchAsync`](https://github.com/loic-sharma/BaGet/blob/controller-refactor/src/BaGet.Protocol/Search/SearchClient.cs#L45) accepts a [`SearchRequest`](https://github.com/loic-sharma/BaGet/blob/controller-refactor/src/BaGet.Protocol/Search/SearchRequest.cs) and returns a [`SearchResponse`](https://github.com/loic-sharma/BaGet/blob/controller-refactor/src/BaGet.Protocol/Search/SearchResponse.cs). Notice that the client and service reuse the exact same interfaces and models.

Reply:
For the first one the PR already does the refactoring for existing V3 search client, see: https://github.com/NuGet/NuGet.Services.Metadata/pull/618/files#diff-066ba730cfa64b581d1b955f35f6085fR60 
I did the bare minimum here so as to not modify existing relevancy tests and break it given that those can be fickle.

For second, I like the idea. But, I would rather not bloat up this PR with those refactorings.